### ISSUE TITLE
width: "fit-content" is not supported by <TabList>

### ISSUE THREAD

**Issue by alanreay** [adobe/react-spectrum#3952]
# 🐛 Bug Report 

Trying to apply “fit-content” to a <TabList> component and getting some odd behavior.  Using the following:
```
<TabList
   marginBottom="size-200"
   width="fit-content"
>
```
When this is rendered though, the <TabList> element has the following styles set:
```
element.style {
    margin-bottom: var(--spectrum-global-dimension-size-200, var(--spectrum-alias-size-200));
    width: var(--spectrum-global-dimension-fit-content, var(--spectrum-alias-fit-content));
}
```
And neither `var(--spectrum-global-dimension-fit-content)` nor `var(--spectrum-alias-fit-content)` are defined anywhere, so no fit-content style is applied.

If I apply fit-content using UNSAFE_style={{ max-width: "fit-content" }} it applies the style, but then has some odd behavior.
When initially rendered, it has the default horizontal tab list as expected.  But if the screen size is reduced enough, it changes to a drop down style element. (I assume as expected).   But if the screen sized is increased, it doesn't return to the horizontal tab list style, it stays as a drop down list.

## 🤔 Expected Behavior

fit-content should be supported on the min-width, max-width, and width properties
As screen size changes it should correctly switch between the minimized drop down style and the horizontal tab list style.

## 😯 Current Behavior

No "fit-content" style is applied to the width.
If applied through UNSAFE_style, it's doesn't correctly switch between the dropdown style and the horizontal tab list

## 💻 Code Sample

```
<Tabs aria-label={"title"}>
	<TabList
		marginBottom="size-200"
		marginStart="size-200"
		marginEnd="size-200"
		width="fit-content"
	>
		<Item key={"general"}>
			{"General settings"}
		</Item>
		<Item key={"builder"}>
			{"Build your report"}
		</Item>
		<Item key={"destinations"}>
			{"Report destination"}
		</Item>
		<Item key={"reportOptions"}>
			{"Report options"}
		</Item>
		<Item key={"scheduling"}>
			{"Scheduling options"}
		</Item>
		<Item key={tabKeys.notification}>
			{"Notification email"}
		</Item>
	</TabList>
	<TabPanels>
		<Item key={"general"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
		<Item key={"builder"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
		<Item key={"destinations"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
		<Item key={"reportOptions"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
		<Item key={"scheduling"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
		<Item key={"notification"}>
			<TabContainer>
				< />
			</TabContainer>
		</Item>
	</TabPanels>
</Tabs>
```
## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |  "@adobe/react-spectrum": "^3.23.0"
| Browser          |  Chrome Version 109.0.5414.87 (Official Build) (x86_64) 
| Operating System |  MacOS v12.6

## 🧢 Your Company/Team

Adobe Analytics GTO

**Comment by LFDanLu** [comment]
We'll want to have these props that accept spectrum values behave in the following manner:
1. Try to resolve the provided value to a spectrum value (e.g. size-2000)
2. If it doesn't, we just pass through the provided value.

**Comment by carterworks** [comment]
+1
This also applies to `<Button>`.

**Comment by snowystinger** [comment]
yes, it applies to everything right now. the line that @LFDanLu shared would enable it across every component supporting size values

**Comment by snowystinger** [comment]
revisiting
TabList has to have the full available width in order for the render for measurement to work.
This issue is a duplicate of [Tabs: zooming out causes TabList to collapse](https://github.com/adobe/react-spectrum/issues/2554)
I'm going to close it for the same reason
